### PR TITLE
Adding script to check for non-plugin pulpcore imports

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #4894

### CODE REVIEW

File: templates/travis/.travis/check_pulpcore_imports.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+set -euv
+
+# exlude tests and imports from pulpcore.plugin
+grep "from pulpcore" **/*.py | grep -v "tests\|plugin"
+
+if [ $? -eq 1]; then
+  exit 0
+else
+  exit 1
+fi
```
Comment (daviddavis) [inline templates/travis/.travis/check_pulpcore_imports.sh.j2:14]: My bash fu is weak :cry:

File: templates/bootstrap/plugin_name/app/viewsets.py.j2 [hunk 2]
```diff
@@ -11,7 +11,7 @@ from rest_framework import status
 from rest_framework.decorators import action
 from rest_framework.response import Response
 
-from pulpcore.app.viewsets import RemoteFilter
+from pulpcore.plugin.viewsets import RemoteFilter
```
Comment (daviddavis) [inline templates/bootstrap/plugin_name/app/viewsets.py.j2:14]: :smile:

Comment (daviddavis) [reply templates/bootstrap/plugin_name/app/viewsets.py.j2:14]: Looks like it's missing from the plugin API. Fix: https://github.com/pulp/pulpcore/pull/666

File: templates/travis/.travis/check_pulpcore_imports.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+{% include 'header.j2' %}
+
+set -uv
+
+# check for imports not from pulpcore.plugin. exclude tests
+MATCHES=$(grep -r --include \*.py "from pulpcore.*import" . | grep -v "tests\|plugin")
```
Comment (mdellweg) [inline templates/travis/.travis/check_pulpcore_imports.sh.j2:8]: I suspect this command to present all findings collapsed in a single line.
If you add a `-n` to the first `grep`, it should report the line-number next to the filename, where it found the line.

File: templates/travis/.travis/check_pulpcore_imports.sh.j2 [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+if [ $? -eq 1 ]; then
```
Comment (mdellweg) [inline templates/travis/.travis/check_pulpcore_imports.sh.j2:10]: If you invert that expression, you would not need an extra branch for `exit 0`.

File: templates/travis/.travis/check_pulpcore_imports.sh.j2 (hunk 3 above)
Comment (daviddavis) [reply templates/travis/.travis/check_pulpcore_imports.sh.j2:8]: Will add `-n`. It does seem to report $MATCHES on newlines when I use echo double quotes:

```
ERROR: Detected bad imports from pulpcore:
./pulp_rpm/app/modulemd.py:4:from pulpcore.app.models.content import Artifact
./pulp_rpm/app/modelresource.py:1:from pulpcore.app.modelresource import QueryModelResource

Plugins should import from pulpcore.plugin.                                                                                                                                                                                    
```

**Pull request merged**